### PR TITLE
feat: rework grid layout update

### PR DESCRIPTION
**Pull request by nastita**

### CODE REVIEW

File: types/grid.ts [hunk 1]
```diff
@@ -31,3 +41,53 @@ export enum WidgetType {
...
   IFRAME = 'IFRAME',
 }
+
+export const GRID_WIDGET_TYPES_SIZES: GridWidgetTypeSizes = {
+  [GridWidgetType.TITLE_LINK]: {
+    [GridWidgetSize.SMALL]: { w: 1, h: 5 },
+    [GridWidgetSize.MEDIUM]: { w: 2, h: 5 },
+    [GridWidgetSize.LARGE]: { w: 3, h: 5 },
+    [GridWidgetSize.FULL]: { w: 4, h: 5 }
+  },
```
Comment (dzbo) [inline types/grid.ts:51]: I wonder if this spectrum of sizes is not limiting us to much, I would consider more scaleable solution based on the WxH ratio:

```
[GridWidgetType.TITLE_LINK]: ['1x5', '2x5', '3x5'...]
```

This way we can easily add other ratios if needed for instance `1x1`, `1x3`...

**Review by dzbo — COMMENTED** [review]
A suggestion for defining sizes in different manner.

File: types/grid.ts (hunk 1 above)
Comment (nastita) [reply types/grid.ts]: agree, this has been updated

File: components/GridView.vue [hunk 2]
```diff
@@ -6,9 +6,13 @@ import {
...
-const COL_NUM_LARGE = 4
-const COL_NUM_SMALL = 2
-const ROW_HEIGHT = 20
+import { toGridLayoutItems } from '@/utils/gridLayout'
+
+import type { GridLayoutItem } from '@/types/grid'
+
+const COL_NUM_LARGE = 2
+const COL_NUM_SMALL = 1
+const ROW_HEIGHT = 280
```
Comment (dzbo) [inline components/GridView.vue:15]: What is the unit here?

Comment (nastita) [reply components/GridView.vue]: renamed var for clarity

File: utils/gridLayout.ts [hunk 3]
```diff
@@ -94,3 +53,77 @@ export function isValidLayout(layout: Widget[]): boolean {
 
   return true
 }
+
+export function toGridLayoutItems(
+  grid: LSP27TheGrid,
+  gridColumns = 4
```
[the remark this answers is not in the document]
Comment (nastita) [reply utils/gridLayout.ts]: updated

**Review by dzbo — APPROVED** [review]

**Pull request merged**